src/schema.rs

```diff
@@ -142,17 +154,28 @@ impl TryCmd {
                         .iter()
                         .find(|s| s.id.as_deref() == Some(id))
                         .expect("id is valid");
-                    let line_nums = step
+                    let mut line_nums = step
                         .expected_stdout_source
                         .clone()
                         .expect("always present for .trycmd");
-                    let mut stdout = stdout.render().expect("at least Text");
-                    // Add back trailing newline removed when parsing
-                    stdout.push('\n');
+
                     let raw = std::fs::read_to_string(path)
                         .map_err(|e| format!("Failed to read {}: {}", path.display(), e))?;
                     let mut normalized = snapbox::utils::normalize_lines(&raw);
+
+                    overwrite_trycmd_status(
+                        exit,
+                        step.expected_status,
+                        step.expected_status_source,
+                        &mut line_nums,
+                        &mut normalized,
+                    )?;
+
+                    let mut stdout = stdout.render().expect("at least Text");
+                    // Add back trailing newline removed when parsing
+                    stdout.push('\n');
```

Comment:
I feel like these are handling things at the wrong place

Reply:
I'm not sure what you mean by that, it was originally there for the stdout overwrite, I just moved it down with the other stdout code

Reply:
Ah, missed that.